src/WebJobs.Extensions.DurableTask/LocalGrpcListener.cs

```diff
@@ -120,18 +137,46 @@ public async Task StopAsync(CancellationToken cancelToken = default)
             }
         }
 
-        private int GetRandomPort()
+        private int GetLikelyAvailablePort()
         {
-            // Get a random port that has not already been attempted so we don't waste time trying
-            // to listen to a port we know is not free.
+            // Get an available port for use in the gRPC server. Try 4001 first, then select a random open port
+            // in the 30000-31000 range.
+            if (this.IsTcpPortFree(DefaultPort))
+            {
+                return DefaultPort;
+            }
+
+            int numAttempts = 50;
+
             int randomPort;
-            do
+            for (int i = 0; i < numAttempts; i++)
             {
                 randomPort = this.portGenerator.Next(MinPort, MaxPort);
+                if (this.IsTcpPortFree(randomPort))
+                {
+                    return randomPort;
+                }
             }
-            while (this.attemptedPorts.Contains(randomPort));
 
-            return randomPort;
+            throw new Exception(string.Format("Failed to get free port for local gRPC server after {0} attempts", numAttempts));
```

Comment:
nit: it's more conventional to do the following:
```suggestion
            throw new InvalidOperationException($"Failed to get free port for local gRPC server after {numAttempts} attempts");
```

Reply:
Done